source/slang/compiler.cpp

```diff
@@ -419,6 +419,19 @@ namespace Slang
             break;
         }
 
+        // Some of the `D3DCOMPILE_*` constants aren't available in all
+        // versions of `d3dcompiler.h`, so we define them here just in case
+        #ifndef D3DCOMPILE_ENABLE_UNBOUNDED_DESCRIPTOR_TABLES
+        #define D3DCOMPILE_ENABLE_UNBOUNDED_DESCRIPTOR_TABLES (1 << 20)
+        #endif
+
+        #ifndef D3DCOMPILE_ALL_RESOURCES_BOUND
+        #define D3DCOMPILE_ALL_RESOURCES_BOUND (1 << 21)
+        #endif
+
+        flags |= D3DCOMPILE_ENABLE_STRICTNESS;
+        flags |= D3DCOMPILE_ENABLE_UNBOUNDED_DESCRIPTOR_TABLES;
```

Comment:
The warning that was generated for having unbounded array was irritating, but did I suppose expose an issue around the use of any capture. It seemed to imply it was better to provide a range. Just getting rid of the warning is certainly the simplest solution, but I wonder if this flag should be controllable via a flag from our API.